x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/notifications/DataFrameAnalyticsAuditMessage.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.notifications;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xpack.core.common.notifications.AbstractAuditMessage;
+import org.elasticsearch.xpack.core.common.notifications.Level;
+import org.elasticsearch.xpack.core.ml.job.config.Job;
+
+import java.util.Date;
+
+public class DataFrameAnalyticsAuditMessage extends AbstractAuditMessage {
+
+    private static final ParseField JOB_ID = Job.ID;
+    public static final ConstructingObjectParser<DataFrameAnalyticsAuditMessage, Void> PARSER =
+        createParser("ml_analytics_audit_message", DataFrameAnalyticsAuditMessage::new, JOB_ID);
+
+    public DataFrameAnalyticsAuditMessage(String resourceId, String message, Level level, Date timestamp, String nodeName) {
+        super(resourceId, message, level, timestamp, nodeName);
+    }
+
+    @Override
+    public final String getJobType() {
+        return "ml_analytics";
```

Comment:
I think this is too broad for the job type.  Since it's in the ML plugin it's effectively the same as using just `analytics`, and that will inevitably hurt us in the future when we want to introduce some other type of analytics that doesn't operate on a 2 dimensional matrix of numbers.

If we are going to stick with the term "data frame" then it should be `data_frame_analytics`.  If not then `ml` here should be replaced by whatever word is going to replace `data_frame`.  And if we want to defer that decision then I'd prefer the name used for the time being is something like `todo_think_of_a_name_before_7.5_analytics` rather than `ml_analytics`, because otherwise we'll release 7.5 with `ml_analytics` and then regret it in the future.

Reply:
For now, I reverted to `data_frame_analytics`. 
This also serves the "think of a name" purpose as if we are to get rid of "data_frame" prefix in other places, this one will need to be updated as well.